File: Python/Garage.py
# Bismillah
# Saya Themy Sabri Syuhada dengan NIM 2203903. Demi keberkahan-Nya, saya berjanji mengerjakan latihan praktikum 4 DPBO dengan jujur dan tidak melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.

class Garage:
    def __init__(self, name, area):
        self.__name = name
        self.__area = area
        self.__vehicles = []

    def add_vehicle(self, vehicle):
        self.__vehicles.append(vehicle)

    def get_vehicles(self):
        return self.__vehicles

    def get_name(self):
        return self.__name

    def set_name(self, name):
        self.__name = name

    def get_area(self):
        return self.__area

    def set_area(self, area):
        self.__area = area

File: CPP/Garage.cpp
#pragma once

#include <iostream>
#include <string>
#include <vector>
#include "Vehicle.cpp"

using namespace std;

class Garage
{
private:
    std::string name;
    double area;
    std::vector<Vehicle *> vehicles;

public:
    // Constructor
    Garage(std::string name, double area)
        : name(name), area(area) {}

    // Method to add a vehicle to the garage
    void add_vehicle(Vehicle *vehicle)
    {
        vehicles.push_back(vehicle);
    }

    // Method to get the list of vehicles in the garage
    std::vector<Vehicle *> get_vehicles()
    {
        return vehicles;
    }

    // Getter for garage name
    std::string get_name()
    {
        return name;
    }

    // Setter for garage name
    void set_name(std::string new_name)
    {
        name = new_name;
    }

    // Getter for garage area
    double get_area()
    {
        return area;
    }

    // Setter for garage area
    void set_area(double new_area)
    {
        area = new_area;
    }
};